File: application/src/Caravelo/Bundle/ApiBundle/BusinessCase/GetSurveysBusinessCase.php
<?php

namespace Caravelo\Bundle\ApiBundle\BusinessCase;

use Caravelo\Bundle\ApiBundle\Component\SurveyMatchTagFilter;
use Caravelo\Bundle\ApiBundle\Model\SurveyDto;
use Caravelo\Bundle\ApiBundle\Model\TagDto;
use Caravelo\Modules\Survey\Domain\Service\SurveyService;

class GetSurveysBusinessCase
{
    /**
     * @var SurveyService
     */
    private $surveyService;
    /**
     * @var SurveyMatchTagFilter
     */
    private $surveyMatchTagFilter;

    public function __construct(
        SurveyService $surveyService,
        SurveyMatchTagFilter $surveyMatchTagFilter)
    {
        $this->surveyService = $surveyService;
        $this->surveyMatchTagFilter = $surveyMatchTagFilter;
    }

    /**
     * @param bool $tagSlug
     * @return \Caravelo\Bundle\ApiBundle\Model\SurveyDto[]
     */
    public function getSurveys($tagSlug = false)
    {
        $surveys = $this->surveyService->getSurveys();
        $surveysDto = [];

        foreach ($surveys as $survey) {
            $surveyDto = new SurveyDto();
            $surveyDto->parseSurvey($survey);
            $surveysDto[] = $surveyDto;
        }

        if ($tagSlug != false) {

            $surveysDto = $this->surveyMatchTagFilter->filter($surveysDto, $tagSlug);
        }

        return $surveysDto;
    }
}

File: application-java/src/main/java/iruiz/app/businesscase/GetSurveysBusinessCase.java
package iruiz.app.businesscase;

import java.util.ArrayList;
import java.util.Iterator;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import iruiz.app.dto.SurveyDto;
import iruiz.app.factory.SurveyDtoFactory;
import iruiz.survey.domain.model.Survey;
import iruiz.survey.domain.service.SurveyService;

@Component
public class GetSurveysBusinessCase {
	
	private SurveyService surveyService;
	private SurveyDtoFactory surveyDtoFactory;
	
	public GetSurveysBusinessCase(){
		
	}
	
	@Autowired
	public void injectSurveyService(SurveyService surveyService){
		this.surveyService = surveyService;
	}
	
	@Autowired
	public void injectSurveyDtoFactory(SurveyDtoFactory surveyDtoFactory){
		this.surveyDtoFactory = surveyDtoFactory;
	}
	
	public ArrayList<SurveyDto> getSurveys(){
		ArrayList<Survey> surveys = this.surveyService.getSurveys();
		ArrayList<SurveyDto> surveysDto = null;
		
		for(Iterator<Survey> i = surveys.iterator(); i.hasNext(); ){
			SurveyDto surveyDto = this.surveyDtoFactory.makeFromSurveyModel(i.next()); 
			surveysDto.add(surveyDto);
		}
		
		return surveysDto;
		
	}
}
